inference-engine/thirdparty/clDNN/kernel_selector/core/cl_kernels/scatter_update_ref.cl

```diff
@@ -15,14 +15,138 @@
 
 #include "include/include_all.cl"
 
-#define GET_UPDATES_INDEX(prefix, idx_order) CAT(prefix, _GET_INDEX)(idx_order)
+#define AXIS_B (0)
+#define AXIS_F (1)
+#define AXIS_W (2)
+#define AXIS_Z (OUTPUT_DIMS - 3)
+#define AXIS_Y (OUTPUT_DIMS - 2)
+#define AXIS_X (OUTPUT_DIMS - 1)
+
 #define GET_OUTPUT_INDEX(idx_order) OUTPUT_GET_INDEX(idx_order)
+
 #if OUTPUT_DIMS == 4
     #define ORDER b,f,y,x
+    #if AXIS_VALUE == AXIS_B
+        #define UPDATES_X_PITCH OUTPUT_X_PITCH
```

Comment:
As I understand, you use pitches from the output instead of intput2 as input2 can be reordered and reshaped. But this works only if output and input2 tensors doesn't have paddings. Please add a check in `Validate` method that `!output.PitchesDifferFromLogicalDims() && !inputs[2].PitchesDifferFromLogicalDims()`

Reply:
Applied.